spring-cloud-dataflow-deployer-cloudfoundry/src/main/java/org/springframework/cloud/dataflow/module/deployer/cloudfoundry/ApplicationModuleDeployer.java

```diff
@@ -317,28 +318,16 @@ private ModuleStatus notRunningInstancesStatus(ModuleDeploymentId id, CloudAppli
 	private Map<String, String> toEnvironmentVariables(HashMap<String, String> args) {
 
 		Map<String, String> env = new HashMap<>(args.size());
-		StringBuilder sb = new StringBuilder();
-		for (Map.Entry<String, String> entry : args.entrySet()) {
-			String oneArg = "--" + entry.getKey() + "=" + quoteValue(entry.getValue());
-			sb.append(bashEscape(oneArg)).append(' ');
+		ObjectMapper objectMapper = new ObjectMapper();
+		try {
+			env.put("SPRING_APPLICATION_JSON", objectMapper.writeValueAsString(args));
```

Comment:
@ericbottard , looks cool :+1: 

But would you mind to explain who reads this variable from the EVN and parse that JSON to the proper `PropertySource`?

Maybe some inline comment here would be good one to have...

Thanks in advance!